### PR TITLE
Add docs for loading model from checkpoint

### PR DESCRIPTION
**Pull request by guarin**
[Load Model from Checkpoint — lightly 1.2.23 documentation.pdf](https://github.com/lightly-ai/lightly/files/9094921/Load.Model.from.Checkpoint.lightly.1.2.23.documentation.pdf)

### CODE REVIEW

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    resnet = lightly.models.ResNetGenerator(name=model_name, width=model_width)
+    model = torch.nn.Sequential(
+        lightly.models.batchnorm.get_norm_layer(3, 0),
+        *list(resnet.children())[:-1],
+    )
+    try:
+        model.load_state_dict(state_dict)
+    except RuntimeError:
+        raise RuntimeError(
+            f'It looks like you tried loading a checkpoint from a model that is not a {model_name} with width={model_width}! '
```
Comment (philippmwirth) [inline docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:23]:
```suggestion
            f'It looks like you tried loading a checkpoint from a model that is not a {model_name} with width={model_width}!\n'
```

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            f'Please set model_name and model_width to the lightly.model.name and lightly.model.width parameters from the '
+            f'configuration you used to run Lightly. The configuration from a Lightly worker run can be found in output_dir/config/config.yaml'
```
Comment (philippmwirth) [inline docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:25]:
```suggestion
            f'configuration you used to run Lightly.\nThe configuration from a Lightly worker run can be found in output_dir/config/config.yaml'
```

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+from collections import OrderedDict
+import torch
+import lightly
+
+def load_ckpt(ckpt_path, model_name='resnet-18', model_width=1):
+    ckpt = torch.load(ckpt_path, map_location='cpu')
```
Comment (philippmwirth) [inline docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:6]: add the `map_location` as an argument above?

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    
+    state_dict = OrderedDict()
+    for key, value in ckpt['state_dict'].items():
+        if ('projection_head' in key) or ('backbone.7' in key):
```
Comment (philippmwirth) [inline docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:10]: can you add a comment here why we're doing this?

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            continue
+        state_dict[key.replace('model.backbone.', '')] = value
+    
+    resnet = lightly.models.ResNetGenerator(name=model_name, width=model_width)
+    model = torch.nn.Sequential(
+        lightly.models.batchnorm.get_norm_layer(3, 0),
```
Comment (philippmwirth) [inline docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:16]: Hm shouldn't the batchnorm be part of the resnet?

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        *list(resnet.children())[:-1],
```
Comment (philippmwirth) [inline docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:17]: The `[:-1]` simply removes the adaptive average pooling, no?

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+image_batch = torch.rand(16, 3, 224, 224)
+out = model(image_batch)
+print(out.shape) # prints: torch.Size([16, 512, 28, 28])
+
+
+# creating a classifier from the pre-trained model
+num_classes = 10
+classifier = torch.nn.Sequential(
+    model,
+    torch.nn.AdaptiveAvgPool2d(1),
```
Comment (philippmwirth) [inline docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:43]: I personally would keep the adaptive average pooling in the code above then you don't need to add it down here again but I'll leave it to you

File: docs/source/docker/advanced/load_model_from_checkpoint.rst [hunk 8]
```diff
@@ -0,0 +1,12 @@
+.. _load-model-from-checkpoint:
+
+Load Model from Checkpoint
+==========================
+
+The Lightly worker can be used to :ref:`train a self-supervised model on your data. <training-a-self-supervised-model>`
+Lightly saves the weights of the model after training to a checkpoint file in
+:code:`output_dir/lightly_epoch_1.ckpt`. This checkpoint can then be further
```
Comment (philippmwirth) [inline docs/source/docker/advanced/load_model_from_checkpoint.rst:8]:
```suggestion
:code:`output_dir/lightly_epoch_X.ckpt`. This checkpoint can then be further
```

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py (hunk 6 above)
Comment (guarin) [reply docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:17]: It removes the classification head from the resnet.

File: docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py (hunk 5 above)
Comment (guarin) [reply docs/source/docker/advanced/code_examples/load_model_from_checkpoint.py:16]: No, sadly not. Not sure why the extra batchnorm was added though.

**Review by philippmwirth — APPROVED** [review]
feel free to simply resolve my commit suggestions
